openjfx/samples/Ensemble8/build.gradle

```diff
@@ -0,0 +1,134 @@
+ *    from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+apply plugin:'application'
+
+repositories {
+    mavenCentral()
+}
+
+mainClassName = "ensemble.EnsembleApp"
+
+def FileCollection apachecp = files(
+   "./lib/lucene-core-7.1.0.jar",
+   "./lib/lucene-grouping-7.1.0.jar",
+   "./lib/lucene-queryparser-7.1.0.jar")
+
+sourceSets {
+    main {
+        java {
+            srcDirs 'src/app/java',
+                    'src/generated/java',
+                    'src/samples/java'
+        }
+        resources {
+            srcDirs 'src/app/resources',
+                    'src/generated/resources',
+                    'src/samples/resources'
+        }
+        // compileClasspath += apachecp
+    }
+}
+def getOSName() {
+    final String osName = System.getProperty("os.name").toLowerCase();
+    if (osName.contains("linux")) {
+        return ("linux");
+    } else if (osName.contains("mac os x") || osName.contains("darwin") || osName.contains("osx")) {
+        return ("mac");
+    } else if (osName.contains("windows")) {
+        return ("win");
+    }
+    return ("");
+}
+
+ext.platform = getOSName()
+
+dependencies {
+    compile "org.openjfx:javafx-controls:11-ea+19:${platform}"
+    compile "org.openjfx:javafx-graphics:11-ea+19:${platform}"
+    compile "org.openjfx:javafx-base:11-ea+19:${platform}"
+    compile "org.openjfx:javafx-fxml:11-ea+19:${platform}"
+    compile "org.openjfx:javafx-web:11-ea+19:${platform}"
+    compile "org.openjfx:javafx-media:11-ea+19:${platform}"
+    compile "org.openjfx:javafx-swing:11-ea+19:${platform}"
+    // compile "org.apache.lucene:lucene-core:7.1.0"
+    compile "org.apache.lucene:lucene-grouping:7.1.0"
+    compile "org.apache.lucene:lucene-queryparser:7.4.0"
+}
+
+
+def excludedArtifacts = configurations.compile.resolvedConfiguration.resolvedArtifacts.findAll {
+println("IT = $it with name $it.name");
+    (it.moduleVersion.id.group == 'org.openjfx' && it.classifier != platform ) || it.name == 'lucene-sandbox'
+}
+
+compileJava {
+    doFirst {
+        classpath -= files(excludedArtifacts.collect { it.file })
+        options.compilerArgs = [
```

Comment:
I get several compilation errors, although they seem to be ignored unless I add the following flag (which the ant-based build scripts in rt/apps/Ensemble do) : 

```
    '-encoding', 'UTF-8',
```

Reply:
added this by default now